databrowser/src/pages/DatasetDetailPage.vue

```diff
@@ -8,10 +8,18 @@
         <Tab>Detail view</Tab>
         <Tab>Raw Data</Tab>
         <div class="absolute right-0">
+          <ButtonPill
+            class="inline-flex md:hidden items-center"
+            @click="showMobileSelect = true"
+          >
+            <span class="pr-2">{{ currentSelected }}</span>
+            <ArrowDown />
+          </ButtonPill>
+
           <ButtonPillGroup
             :data="suportedLanguages"
-            :default-selected="defaultLanguage"
-            :initial-selected="initialSelected"
+            :initial-selected="currentSelected"
+            class="hidden md:inline-flex"
             @selected-change="changeLanguage"
           />
```

Comment:
I would wrap this into a `LanguagePicker` component, to reduce the complexity of the DatasetDetailPage.